http-client/src/main/java/io/micronaut/http/client/netty/DefaultNettyHttpClientRegistry.java

```diff
@@ -425,49 +421,38 @@ public <T> Optional<MessageBodyWriter<T>> findWriter(Argument<T> type, List<Medi
                     }
                 });
             }
-            return client;
+            return builder.build();
         });
     }
 
-    private DefaultHttpClient buildClient(
-            LoadBalancer loadBalancer,
-            HttpVersionSelection httpVersion,
+    private DefaultHttpClientBuilder clientBuilder(
             HttpClientConfiguration configuration,
             String clientId,
-            String contextPath,
             BeanContext beanContext,
             AnnotationMetadata annotationMetadata) {
 
-        EventLoopGroup eventLoopGroup = resolveEventLoopGroup(configuration, beanContext);
-        ConversionService conversionService = beanContext.getBean(ConversionService.class);
         String addressResolverGroupName = configuration.getAddressResolverGroupName();
-        AddressResolverGroup<?> resolverGroup = addressResolverGroupName == null ? null : beanContext.getBean(AddressResolverGroup.class, Qualifiers.byName(addressResolverGroupName));
-        return new DefaultHttpClient(
```

Comment:
Can we create an instance of the builder with scope prototype using the bean context? Maybe a factory or something.

Reply:
The problem is that from the bean context, clients always have to go to the registry for (a) service id configuration (b) service id pooling. It is not actually possible to use the Builder to create a managed (in-context) client, you always have to get a ready-built client from the registry